src/ClientData/TimerChain.cpp

```diff
@@ -28,14 +28,14 @@ orbit_client_data::TimerChain::~TimerChain() {
 }
 
 orbit_client_data::TimerBlock* orbit_client_data::TimerChain::GetBlockContaining(
-    const orbit_client_data::TextBox* element) const {
+    const orbit_client_protos::TimerInfo& element) const {
   orbit_client_data::TimerBlock* block = root_;
   while (block != nullptr) {
     uint32_t size = block->size();
     if (size != 0) {
-      orbit_client_data::TextBox* begin = &block->data_[0];
-      orbit_client_data::TextBox* end = &block->data_[size - 1];
-      if (begin <= element && end >= element) {
+      orbit_client_protos::TimerInfo* begin = &block->data_[0];
+      orbit_client_protos::TimerInfo* end = &block->data_[size - 1];
+      if (begin <= &element && end >= &element) {
```

Comment:
I don't have an easy-fix.
1- Is there a case where it can give us a false positive? Or it is just error prone for the future?
2- Should I create a bug?

Reply:
Yes, the compiler is totally allowed to give us a false positive if `element` is not from the array (meaning `begin <= &element && end >= &element` could be true even though `element` is not from `block->data_`). But it's not likely to happen. Compiler vendors tend to not break their behaviour because they know projects (falsely) rely on it. So since it's working right now, it's unlikely to break in the near future.

Nevertheless we should avoid it - because it's UB and because this can harm optimization. Feel free to create a bug but since it's working and likely to continue to work this shouldn't have a high prio.

Reply:
http://b/194268700